### DISCUSSION TITLE
Few improvement ideas

### DISCUSSION THREAD

**Discussion by Golgautier** [immich-app/immich#3590] [General]
Hi everyone,

First, I would like to congratulate @alextran1502 and all contributers too for this fantastic product. I expect to be able to help to his improvement but I am afraid to be a bit weak regarding my dev level.
I could probably help at infra level (I you want I create a helm chart for Immich deployement on k8s I can do that).

I have some ideas/expectations to improve this product, I put them here, I hope it will interest you. Some of them are probably already mentioned in the past. 

- Toggle to display pictures from shared album into the main gallery
- Ability to change EXIF info (sometimes upload have strange info, it would be interesting to be able to change them with the GUI. Currently I have tu delete the IMMICH image, change the source image, and upload it again). I image rotation is one of the most intresting thing to change in the exif data.
- Face recognition : I would like to be able to manualy change face recognition result in photos. Mainly because of baby pictures, I currently have a people found by Immich, that concentrate 3 different real people. 
- Ability to free space on mobile (remove local image but keep Immich one)
- In my case, I would like to share all my new pictures with my wife, but I do not want to share my old picture from "my previous live". 2 options to do that : define a shared album where all new photos will be pushed, or add a privacy zone, not seen by a partner.

I think the most important ideas I got are here... I will probably have new ones later...

I expect it will "help" 

Thank you again for this great job.

**Comment by alextran1502** [comment]
Hello,

Thanks for the ideas of improvement. We acknowledge all of the points above and those are goals we want to eventually add to Immich as well

**Reply by Golgautier** [reply]
Great !

Let me know if I can help help with my few skills, for anything.

**Comment by geomule** [comment]
I would certainly the idea of allowing EXIF data editing, especially the date. Basically, every time and image is share via Signal messaging or Whatsapp, the date gets modified so when it's uploaded to Immich, it uploaded in the wrong month/year, therefore not correctly showing on the timeline. Its a real hassel to have to export, change date, re-import.

Being able to assign a photo to a location like in digikam, that would be great.

Finally manual tagging.

These are the three things that I like of digikam. I can see Immich becoming my digikam substitute.

**Reply by jrasm91** [reply]
The top two are in the works. There is a metadata writing PR currently open.

**Comment by tomikonieczny** [comment]
Hello,

at the beginning many thanks for the app!

selection improvement is welcome

web:
when selecting pictures it would be great to have option to do it also in full screen preview not losing previous selection (same as in google photos)

mobile app (android):
the same as in web app, additional issue is losing selection while changing screen orientation
